slides_api.py

```diff
@@ -0,0 +1,106 @@
+        drive_service  = build('drive',  'v3', credentials=creds)
+
+        body = {"name": f"{client_name} - Causal Impact Insights"}
+        drive_response = (
+            drive_service.files().copy(fileId=slide_id, body=body).execute()
+        )
+        slide_copy_id = drive_response.get("id")
+        
+        request_data = get_value_session(session_id, 'output_data')
+
+        to_replace = [
+            ['CLIENT_NAME', client_name], 
+            ['DATE', datetime.date.today().strftime("%d-%m-%Y")],
+            ['TARGET_EVENT', request_data['target_event']],
+            ['COVARIATES', request_data['covariates']],
+            ['PREPERIOD-START', request_data['from_date']],
+            ['PREPERIOD-END', (datetime.datetime.strptime(request_data['event_date'], "%Y-%m-%d") - datetime.timedelta(days=1)).strftime("%Y-%m-%d")],
+            ['POSTPERIOD-START', request_data['event_date']],
+            ['POSTPERIOD-END', request_data['to_date']],
+            ['P_VALUE', request_data['p_value']],
+            ['INCR_ACTIONS', request_data['tot_inc']]
+        ]
+        requests = []
+        for label in to_replace:
+            requests.append({
+                'replaceAllText': {
+                    'replaceText': label[1],
+                    'containsText': {
+                        'matchCase': False,
+                        'text': f'{{{{{label[0]}}}}}'
+                    }
+                }
+            })
+
+        page_id = ""
+        final_element = ""
+        presentation = (slides_service.presentations().get(presentationId=slide_copy_id).execute())
+        slides = presentation.get("slides")
+        for slide in slides:
+            elements = slide.get('pageElements')
+            if elements:
+                for element in elements:
+                    if element.get("description") == "CHART":
+                        page_id = slide['objectId']
+                        final_element = element
+                        break;  
+        
+        add_img = {
+             "createImage": {
+                "objectId": "my_id",
+                "url": f"https://storage.googleapis.com/{os.environ.get('IMAGE_BUCKET')}/{request_data['image_name']}",
+                "elementProperties": {
+                    "pageObjectId": page_id,
+                    'size': {
+                         
+                    },
+                    'transform': {
+                        
+                    },
+                },
+            }
+        }
+        add_img['createImage']['elementProperties']['size'] = final_element['size']
+        add_img['createImage']['elementProperties']['transform'] = final_element['transform']
+        requests.append(add_img)
+
+        response = slides_service.presentations().batchUpdate(
+            body={'requests': requests},
+            presentationId=slide_copy_id
+        ).execute()
+
+        return slide_copy_id
+
+    except:
+        print("Slides error")
```

Comment:
no error logging here meaning we will never know if it fails

Reply:
Fixed